src/main/surfrdMod.F90

```diff
@@ -780,20 +780,14 @@ subroutine surfrd_veg_all(begg, endg, ncid, ns, actual_numcft)
 
     ! Check the file format for CFT's and handle accordingly
     if ( actual_numcft > 0 ) then
-       ! Full crop on file and in CLM
-       if ( create_crop_landunit .and. actual_numcft > 2 .and. use_crop)then
-          call surfrd_cftformat( ncid, begg, endg, wt_cft, fert_cft, cft_size, natpft_size )  ! Format where CFT's is read in a seperate landunit
-       ! Generic crop in file and generic crop in model
-       else if ( create_crop_landunit .and. actual_numcft == 2 .and. cft_size == 2 )then
-          call surfrd_cftformat( ncid, begg, endg, wt_cft, fert_cft, cft_size, natpft_size )  ! Format where CFT's is read in a seperate landunit
-       ! Full crop in file, but model is only with generic crop
-       else if ( create_crop_landunit .and. actual_numcft > 2 .and. .not. use_crop)then
+       if ( create_crop_landunit )then
+          ! Cases: Full crop in file and in model
+          !        Generic crop in file and in model
+          !        Full crop in file, generic crop in model
           cftsize = actual_numcft
           allocate(array2DCFT (begg:endg,cft_lb:cftsize-1+cft_lb))
           allocate(array2DFERT(begg:endg,cft_lb:cftsize-1+cft_lb))
           call surfrd_cftformat( ncid, begg, endg, array2DCFT, array2DFERT, cftsize, natpft_size )
```

Comment:
Removed call collapse_crop_types from here, because the same call a few lines down now occurs for the combination of cases, leading to the same result.